### PR TITLE
Remove `child-workspace-package-names-as-json` script in favour of `yarn workspaces filter` command

### PR DESCRIPTION
**Pull request by Mrtenz**
This adds a new Yarn plugin, which replaces the `child-workspace-package-names-as-json` script. The plugin introduces a new command: `yarn workspaces filter`:

```
━━━ Usage ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

$ yarn workspaces filter

━━━ Options ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  --include #0    List workspaces based on a glob pattern
  --exclude #0    Exclude workspaces based on a glob pattern
  --json          Format the output as a JSON array

━━━ Details ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

This command will filter workspaces based on the given criteria. It's like `yarn 
workspaces list` but on steroids.

━━━ Examples ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

List workspaces based on a glob pattern
  $ yarn workspaces filter --include "packages/*"

Exclude workspaces based on a glob pattern
  $ yarn workspaces filter --exclude "packages/*/foo"
```

This lets us use a glob pattern for including or excluding Yarn workspaces.

Prerequisite for #1488.

### CODE REVIEW

File: .yarn/patches/@types-glob-npm-7.1.4-d45247eaa2.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
++    interface IOptions extends minimatch.MinimatchOptions {
+         cwd?: string | undefined;
+         root?: string | undefined;
+         dot?: boolean | undefined;
+@@ -69,7 +69,7 @@ declare namespace G {
+     }
+ 
+     interface IGlobBase {
+-        minimatch: minimatch.IMinimatch;
++        minimatch: minimatch.MinimatchOptions;
```
Comment (FrederikBolding) [inline .yarn/patches/@types-glob-npm-7.1.4-d45247eaa2.patch:19]: ?

File: .yarn/plugins/local/plugin-workspaces-filter.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        // load the plugin, even if the command is not used, and `minimatch`
+        // may not be installed.
+        const { minimatch } = await import('minimatch');
+
+        const configuration = await Configuration.find(
+          this.context.cwd,
+          this.context.plugins,
+        );
+        const { project } = await Project.find(configuration, this.context.cwd);
+        const { workspaces } = project;
```
Comment (FrederikBolding) [inline .yarn/plugins/local/plugin-workspaces-filter.js:56]: Is workspaces only the "root" workspaces here or does it recurse through all workspaces including workspaces within workspaces?

File: .yarn/patches/@types-glob-npm-7.1.4-d45247eaa2.patch (hunk 1 above)
Comment (Mrtenz) [reply .yarn/patches/@types-glob-npm-7.1.4-d45247eaa2.patch:19]: `@types/glob` assumes we use `@types/minimatch`, but `minimatch` has its own types, which does not have `IMinimatch`.

File: .yarn/plugins/local/plugin-workspaces-filter.js (hunk 2 above)
Comment (Mrtenz) [reply .yarn/plugins/local/plugin-workspaces-filter.js:56]: This recurses through all workspaces. This lets us specify just the root workspaces using `packages/*`, or everything using `packages/**` for example.

Comment (FrederikBolding) [reply .yarn/plugins/local/plugin-workspaces-filter.js:56]: 👍

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**